src/games/strategy/engine/lobby/client/ui/LobbyGameTableModel.java

```diff
@@ -121,10 +123,18 @@ private void updateGame(final GUID gameId, final GameDescription description) {
     SwingUtilities.invokeLater(new Runnable() {
       @Override
       public void run() {
-        Tuple<GUID,GameDescription> toReplace = findGame(gameId);
-        int replaceIndex = gameList.indexOf(toReplace);
-        gameList.set(replaceIndex, Tuple.of(gameId, description));
-        fireTableRowsUpdated(replaceIndex, replaceIndex);
+        if (gameId == null) {
+          return;
+        }
+
+        final Tuple<GUID, GameDescription> toReplace = findGame(gameId);
+        if (toReplace == null) {
+          gameList.add(Tuple.of(gameId, description));
```

Comment:
Shouldn't it call the fireTableRowsInserted method here as well since it did that in the previous add method?